### ISSUE TITLE
[DISTANCE] LOW: UpdateDistanceRequest 1:1:1 - UpdateMoneyRequestDescription - Split out into its own command

### ISSUE THREAD

**Issue by tgolen** [Expensify/App#32314]
There will be three pieces to this:
- Auth: Using the logic from `EditMoneyRequest`, create a new command called `UpdateMoneyRequestDescription` following the example in https://github.com/Expensify/Auth/pull/8986
- Web-E: Add a new API endpoint called `UpdateMoneyRequestDescription` following the example in https://github.com/Expensify/Web-Expensify/pull/39445
- App: Implement the new API command following the example in https://github.com/Expensify/App/pull/30737
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0166ac586517ecf2dd</li>
        <li>Upwork Job ID: 1730346767459143680</li>
        <li>Last Price Increase: 2023-11-30</li>
    </ul>
</details>

**Comment by chiragsalian** [comment]
Handing this over to @NikkiWines. I have an auth PR in draft and you can continue from it. it seems to work fine in my testing. I tested it via a test.php file having,
```
$params = [
    "authToken" => "<authToken>",
    "transactionID" => "2390656467468356405", // a transaction the user has access to
    "modifiedExpenseReportActionID" => 8299677307715651178, // a random reportActionID, with every test call increment this by 1.
    "comment" => "test one five",
];

Auth::updateMoneyRequestDescription($params); // create a simple auth updateMoneyRequestDescription method to call the auth endpoint.
```
and confirmed it updates the comment just fine. 
```
sqlite> select comment from transactions where transactionID = 2390656467468356405;
comment
----------------------------------------------
{"comment":"test one five","test":"something"}
``` 
i added the test something to confirm other parts of the comment are not lost and just the comment bit is updated.
the report action looks fine on newDot too.
![image](https://github.com/Expensify/App/assets/14356145/af3171c2-652f-4c64-a03d-1bc7dc83d709)

So things remaining
1. Auth tests
2. Web and newDot code to split it into its own command.

Hit me up on newDot if you have any questions.

**Comment by chiragsalian** [comment]
oh i just remembered another remaining item and i.e., to maybe return or send a push notification of commentJSON so that the client gets the changes.

**Comment by NikkiWines** [comment]
Planning on tackling this during the week

**Comment by NikkiWines** [comment]
[Auth PR](https://github.com/Expensify/Auth/pull/9385) is ready for review, will follow up with Web-E and App PRs tomorrow

**Comment by NikkiWines** [comment]
Auth PRs been updated, Web-E and App PRs are WIPs!

**Comment by NikkiWines** [comment]
Noting here that we'll need to remove [this comment](https://github.com/Expensify/Auth/blob/ba8973367a13f66b7ac6fae3009c27cbc245ac5c/auth/command/UpdateMoneyRequestDescription.cpp#L65C1-L66) before this issue can be fully closed out. @tgolen since all of the `UpdateMoneyRequest<X>` commands will have this same comment, is there an issue that covers updating these instances so they don't stay as perpetual todos in our code?

**Comment by aldo-expensify** [comment]
> is there an issue that covers updating these instances so they don't stay as perpetual todos in our code?

Maybe we should create an issue to deal with all of them together. I'm just guessing that after moving the logic needed from PHP to cpp for one, fixing the others will be trivial.

**Comment by tgolen** [comment]
Ah, that's a good point. I think that comment should be removed from the code and we should create a GH issue to move those updates to Auth at some point. I don't know when it will be valuable to move them.

I agree with Aldo that once the work is done to move the updates for one command, that will be 99% of the work.

**Comment by NikkiWines** [comment]
This is all done!
